Feature/delta education v1.0 (#8343)
microsoftgraph/microsoft-graph-docs: api-reference/v1.0/api/educationclass-delta.md

---
title: "educationClass: delta"
description: "Get newly created or updated classes, including membership changes, without having to perform a full read of the entire class collection."
localization_priority: Normal
author: "mlafleur"
ms.prod: "education"
doc_type: apiPageType
---

# educationClass: delta

Namespace: microsoft.graph

> Get newly created or updated classes, including membership changes, without having to perform a full read of the entire class collection. See [Use delta query](/graph/delta-query-overview) for details.

Reviewer: For more information, see [Use delta query](/graph/delta-query-overview).

Author: ```suggestion
Get newly created or updated classes, including membership changes, without having to perform a full read of the entire class collection. For more information, see [Use delta query](/graph/delta-query-overview).
```

## Permissions

One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

| Permission type                        | Permissions (from least to most privileged)                              |
| :------------------------------------- | :----------------------------------------------------------------------- |
| Delegated (work or school account)     | EduRoster.ReadBasic, EduRoster.Read, or EduRoster.ReadWrite              |
| Delegated (personal Microsoft account) | Not supported.                                                           |
| Application                            | EduRoster.ReadBasic.All, EduRoster.Read.All, or EduRoster.WriteWrite.All |

## HTTP request

<!-- { "blockType": "ignored" } -->

```http
GET /education/classes/{id}/delta
GET /education/me/classes/{id}/delta
```

## Request headers

| Name          | Description   |
| :------------ | :------------ |

> | Authorization | Bearer {code} |

Reviewer: Bearer {token}

Author: fixed, thanks!

> | Authorization | Bearer {code} |

Author: ```suggestion
| Authorization | Bearer {token} |
```

## Request body

Do not supply a request body for this method.

## Response

If successful, this method returns a `200 OK` response code and an [educationClass](../resources/educationclass.md) collection object in the response body.

## Example

> The following example shows how to call this API.

Reviewer: This sentence isn't necessary.

Author: ```suggestion
```

> ##### Request

Reviewer: This should be a level 3 header: '### Request'

Author: ```suggestion
### Request
```

The following is an example of the request.

<!-- {
  "blockType": "request",
  "name": "educationclass_delta"
}-->

```http
GET https://graph.microsoft.com/v1.0/education/classes/{id}/delta
```

> ##### Response

Reviewer: This should be a level 3 header: '### Response'

Author: ```suggestion
### Response
```

The following is an example of the response. 

>**Note:** The response object shown here might be shortened for readability. All the properties will be returned from an actual call.

<!-- {
  "blockType": "response",
  "truncated": true,
  "@odata.type": "microsoft.graph.educationClass",
  "isCollection": true
} -->

```http
HTTP/1.1 200 OK
Content-type: application/json
Content-length: 585

{
  "value": [
    {

>       "displayName": "displayName-value",

Reviewer: These should be real-world looking examples so that the customer knows what format should be expected in the value. For example, "displayName": "contosoClass1",

Author: I copied the page from the beta one. We'd need to fix it across the board and hopefully the content owner would do that.

Reviewer 2: That's a fair comment about real-world examples. Vincent is correct that this should be fixed across the board for all of the Education APIs. I'd recommend letting this go for now, as it is consistent with the rest of the EDU doc, and then we improve the examples all at once for EDU.

>       "displayName": "displayName-value",

Reviewer: These should be real-world examples rather than place holders. Possibly run Graph explorer on a test tenant and use the result to fill in with real looking values.

Author: we already had a conversation about that here and it was decided that specific issue needs to be addressed both in beta and v1 in a separate pull request. https://github.com/microsoftgraph/microsoft-graph-docs/pull/8343/files#r428168765
